deltacat/compute/compactor/compaction_session.py

```diff
@@ -66,6 +67,7 @@ def compact_partition(
         min_hash_bucket_chunk_size: int = 0,
         compacted_file_content_type: ContentType = ContentType.PARQUET,
         delete_prev_primary_key_index: bool = False,
+        schema_on_read: Optional[pa.schema] = None,
```

Comment:
Instead of passing this in, I think the default behavior should be to assign the appropriate schema automatically inside of the storage interface's `download_delta` and `download_delta_manifest_entry` implementations via a call to `get_table_version_schema` (which should also typically maintain an LRU cache of table version schemas to avoid making a call out to a remote service and/or filesystem every time).

Reply:
Sounds cleaner to me!